src/charactersheet/viewmodels/character/spells/index.js

```diff
@@ -1,294 +1,160 @@
+            'damageType asc': { field: 'damageType', direction: 'asc'},
+            'damageType desc': { field: 'damageType', direction: 'desc'},
+            'damage asc': { field: 'damage', direction: 'asc'},
+            'damage desc': { field: 'damage', direction: 'desc'},
+            'level asc': { field: 'level', direction: 'asc', numeric: true},
+            'level desc': { field: 'level', direction: 'desc', numeric: true},
+            'castingTime asc': { field: 'castingTime', direction: 'asc'},
+            'castingTime desc': { field: 'castingTime', direction: 'desc'},
+            'range asc': { field: 'range', direction: 'asc'},
+            'range desc': { field: 'range', direction: 'desc'}
+        };
+    }
+
+    highestLevelSpellSlot = ko.pureComputed(() => {
+        const availableSlots = filter(this.spellSlots(), (spellSlot)=> (spellSlot.used() < spellSlot.max()));
+        const highestSlot = maxBy(availableSlots, (spellSlot) => (spellSlot.level()));
+        if (highestSlot) {
+            return highestSlot.level();
+        }
+        return 0;
+    })
 
-    self.numberOfPrepared = ko.computed(function() {
-        var prepared = 0;
-        self.spellbook().forEach(function(spell) {
-            if (spell.prepared() === true) {
-                prepared++;
+    filteredAndSortedEntities = ko.pureComputed(() =>  {
+        let spellbook = this.entities();
+        if (this.filteredByCastable()) {
+            spellbook = this.entities().filter(this.spellIsCastable, this);
+        }
+        return SortService.sortAndFilter(spellbook, this.sort(), null);
+    }, this);
+
+    toggleSpellFilter = () => {
+        this.filteredByCastable(!this.filteredByCastable());
+    }
+
+    getDefaultSort () {
+        return this.sorts()['level asc'];
+    }
+
+    setUpSubscriptions () {
+        super.setUpSubscriptions();
+        this.subscriptions.push(Notifications.spellslot.changed.add(this.updateSpellSlots, this));
+        this.subscriptions.push(Notifications.spellslot.added.add(this.updateSpellSlots, this));
+        this.subscriptions.push(Notifications.spellslot.deleted.add(this.updateSpellSlots, this));
+        this.subscriptions.push(Notifications.spellstats.changed.add(this.updateSpellStats, this));
+    }
+
+    updateSpellStats = async (stats) => {
+        this.spellStats().importValues(stats.exportValues());
+    }
+
+    updateSpellSlots = async (slot) => {
+        const response = await SpellSlot.ps.list({ coreUuid: this.coreKey });
+        this.spellSlots(response.objects);
+    }
+
+    spellIsCastable = (spell) => {
+        if (spell.level() === 0) {
+            return true;
+        } else if (spell.level() > this.highestLevelSpellSlot()) {
+            return false;
+        } else if (spell.level() === 0) {
+            return true;
+        } else if ( spell.alwaysPrepared() === true) {
+            return true;
+        } else if (spell.prepared() === true) {
```

Comment:
Just because a spell is `prepared` or `alwaysPrepared` doesn't mean it's castable, right?

Reply:
it is castable if the previous returns didn't return false first. See line 101 for a false catch.